crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1791,7 +1791,38 @@ impl<
                 let src_reg = self.storage_manager.load_to_general_reg(&mut self.buf, src);
                 ASM::neg_reg64_reg64(&mut self.buf, dst_reg, src_reg);
             }
-            x => todo!("NumNeg: layout, {:?}", x),
+            LayoutRepr::F32 => {
+                let dst_reg = self.storage_manager.claim_float_reg(&mut self.buf, dst);
+                let src_reg = self.storage_manager.load_to_float_reg(&mut self.buf, src);
+                self.storage_manager
+                    .with_tmp_float_reg(&mut self.buf, |_, buf, tmp| {
+                        // gcc and llvm will do `f ^ 0x8000_0000` to flip the sign bit
+                        // but we are doing 0-f here because I couldn't figure out how
+                        // to load floats into general registers
+                        ASM::mov_freg32_imm32(buf, &mut self.relocs, tmp, 0.0f32);
+                        ASM::sub_freg32_freg32_freg32(buf, dst_reg, tmp, src_reg);
+                    })
+            }
```

Comment:
@folkertdev 

Do you know how to load floats into general registers?

Reply:
I may have been mistaken about that. The equivalent code in C is compiled to use `xmm` registers which are floating point registers. There's just no `xorps` or `xorpd` ops defined yet for gen-llvm (or equivalent "bitwise xor of floating point register" for gen-wasm/gen-dev), but I can work on that.

[godbolt C example](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:___c,selection:(endColumn:2,endLineNumber:7,positionColumn:2,positionLineNumber:7,selectionStartColumn:2,selectionStartLineNumber:7,startColumn:2,startLineNumber:7),source:'/*+Type+your+code+here,+or+load+an+example.+*/%0Adouble+neg_f64(double+f)+%7B%0A++++return+-f%3B%0A%7D%0Afloat+neg_f32(float+f)+%7B%0A++++return+-f%3B%0A%7D'),l:'5',n:'0',o:'C+source+%231',t:'0')),k:39.03462749213011,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:cg142,filters:(b:'0',binary:'1',binaryObject:'0',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:___c,libs:!(),options:'',overrides:!(),selection:(endColumn:30,endLineNumber:7,positionColumn:30,positionLineNumber:7,selectionStartColumn:30,selectionStartLineNumber:7,startColumn:30,startLineNumber:7),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:60.96537250786989,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

Reply:
Worth giving it a shot 😄